cli/cmd/plugin/unmanaged-cluster/cluster/cluster.go

```diff
@@ -41,16 +42,26 @@ type Manager interface {
 	// errors found, otherwise a list of the errors that need to be resolved.
 	// A list of warning messages can be returned that are not blocking errors.
 	PreflightCheck() ([]string, []error)
-	// ProviderNotify returns any provider specific notifications or messages.
+	// PreProviderNotify returns any provider specific notifications or messages
+	// to log before bootstrapping starts.
 	// Each string will be displayed on its own line.
-	ProviderNotify() []string
+	PreProviderNotify() []string
+	// PostProviderNotify returns any provider specific notifications or messages
+	// to log after bootstrapping has finished.
+	// Each string will be displayed on its own line.
+	PostProviderNotify() []string
+	// WithLogFile sets the log file to send provider bootstrapping logs to
+	// should be a fully qualified path
+	WithLogFile(string)
```

Comment:
I wonder if the bootstrapping log file should come in as part of the [config.UnmanagedClusterConfig](https://github.com/jpmcb/community-edition/blob/3ca965fc3b79c7bd410c32c83a3e74ed17f7f0fb/cli/cmd/plugin/unmanaged-cluster/config/config.go#L74-L118).

Here is my thinking/reasoning:

* We should strive to keep the surface area of our Manager interface minimal, since it will impact every provider.
* Adding this to the config would make it configurable by users who wish to send their bootstrapping logs to arbitrary locations.